packages/manager/apps/web/client/app/hosting/hosting.controller.js

```diff
@@ -828,6 +836,70 @@ export default class {
     });
   }
 
+  handleCDNProperties() {
+    return this.HostingCdnSharedService.getCDNProperties(
+      this.$scope.hosting.serviceName,
+    )
+      .then(({ data: cdn }) => {
+        this.$scope.cdnProperties = cdn;
+      })
+      .catch((err) => {
+        this.Alerter.error(err);
+        this.$scope.cdnProperties = null;
+      });
+  }
+
+  /**
+   * This function can be removed once all CDN has been migrated by AGORA
+   * The isUpgradable variable can also removed
+   * @returns {Promise<{}>}
+   */
+  async simulateUpgradeAvailability() {
```

Comment:
using `async` is not a good idea in an angularjs controller due to the promise not being "synced" with the digest cycle

Reply:
Thanks for the info.
I just reuse the existed code.
I will try to find a new way to resolve that.